OIPA/iati_organisation/parser/organisation_2_03.py

```diff
@@ -215,6 +215,48 @@ def iati_organisations__iati_organisation(self, element):
 
             return element
 
+    def iati_organisations__iati_organisation__name(self, element):
+
+        # Although OrganisationName and Organisation has One-to-One relation
+        # on the database level, we check here whether element 'name' occurs
+        # only once in the parent element 'organisation'.
+        organisation = self.get_model('Organisation')
+        if 'OrganisationName' in self.model_store:
+            for name in self.model_store['OrganisationName']:
+                if name.organisation == organisation:
+                    raise ParserError("Organisation", "OrganisationName",
+                                      "must occur no more than once.")
+        # narrative is parsed in different method but as it is required
+        # sub-element in 'name' element so we check it here.
+        narrative = element.xpath("narrative")
+        if len(narrative) < 1:
+            raise RequiredFieldError("OrganisationName", "narrative",
+                                     "must occur at least once.")
+        organisation_name = OrganisationName()
+        organisation_name.organisation = organisation
+
+        self.register_model("OrganisationName", organisation_name)
+        return element
+
+    def iati_organisations__iati_organisation__name__narrative(self, element):
```

Comment:
I think this is essentially correct, but before merging changes, the previous method (`iati_organisations__iati_organisation__name`) should be fixed an then removed from this particular PR (because essentially you want to merge 2 different small features inside 1 PR)